src/common/Utilities/EventMap.cpp

```diff
@@ -94,7 +94,18 @@ uint32 EventMap::ExecuteEvent()
 
 void EventMap::DelayEvents(Milliseconds delay)
 {
-    _time = delay.count() < _time ? _time - delay.count() : 0;
+    if (Empty())
+        return;
+
+    EventStore delayed;
+
+    for (EventStore::iterator itr = _eventMap.begin(); itr != _eventMap.end();)
+    {
+        delayed.insert(EventStore::value_type(itr->first + delay, itr->second));
+        _eventMap.erase(itr++);
+    }
+
+    _eventMap.insert(delayed.begin(), delayed.end());
```

Comment:
Maybe something like this?

```suggestion
    EventStore delayed = std::move(_eventMap);

    for (EventStore::iterator itr = delayed.begin(); itr != delayed.end();)
    {
        EventStore::node_type node = delayed.extract(itr++);
        node.key() = node.key() + delay;
        _eventMap.insert(_eventMap.end(), std::move(node));
    }
```

Reply:
Hah, `extract`. That's a new one. Thanks for the suggestion!